src/app-layer-smtp.c

```diff
@@ -405,6 +405,83 @@ static void SMTPNewFile(SMTPTransaction *tx, File *file)
             smtp_config.content_inspect_min_size);
 }
 
+static int SMTPFileContainerAlloc(SMTPState *smtp_state)
+{
+    /* Make sure file container allocated */
+    if (smtp_state->files_ts == NULL) {
+        smtp_state->files_ts = FileContainerAlloc();
+        if (smtp_state->files_ts == NULL) {
+            SCLogError(SC_ERR_MEM_ALLOC, "Could not create file container");
+            return MIME_DEC_ERR_MEM;
```

Comment:
I further step of separating the various api's would to not use `MIME_DEC...` for returns here.